src/Compilers/Core/Portable/Collections/OrderPreservingMultiDictionary.cs

```diff
@@ -159,9 +163,9 @@ public struct ValueSet : IEnumerable<V>
             /// </summary>
             private readonly object _value;
 
-            internal ValueSet(V value)
+            internal ValueSet([DisallowNull] V value)
             {
-                _value = value;
+                _value = value!;
```

Comment:
@cston is it expected that this `!` is necessary?

Reply:
Logged [https://github.com/dotnet/roslyn/issues/40139](https://github.com/dotnet/roslyn/issues/40139).

---
In reply to: [353897348](https://github.com/dotnet/roslyn/pull/40029#discussion_r353897348) [](ancestors = 353897348)